core/frontend/src/components/version-chooser/VersionChooser.vue

```diff
@@ -0,0 +1,506 @@
+      await back_axios({
+        method: 'get',
+        url: '/version-chooser/v1.0/version/current',
+        timeout: 500,
+      })
+        .then(() => {
+          if (this.waiting) {
+            // Allow 3 seconds so the user can read the "complete" message
+            // reload(true) forces the browser to fetch the page again
+            setTimeout(() => { window.location.reload(true) }, 3000)
+          }
+        })
+        .catch((error) => {
+          console.log(error)
+          this.waiting = true
+        })
+    },
+    fixVersion(version: string) {
+      /** It turned out that our semvers are wrong... oopss
+      This turns 1.0.0.beta12 into 1.0.0-beta.12
+      Additionally filters out tags with no '.' in it, which
+      can be improved
+      */
+      if (version.includes('beta') && !version.includes('-beta.')) {
+        return version.replace('.beta', '-beta.')
+      }
+      if (!version.includes('.')) {
+        return null
+      }
+      return version
+    },
+    isSemVer(version: string): boolean {
+      // validates a version as SemVer compliant
+      try {
+        const fixed_version = this.fixVersion(version)
+        if (fixed_version == null) {
+          return false
+        }
+        const semver = new SemVer(fixed_version)
+        return semver !== null
+      } catch (error) {
+        return false
+      }
+    },
+    sortVersions(versions: string[]) {
+      return versions.sort(
+        (a: string, b: string) => {
+          const ver_a = this.fixVersion(a)
+          const ver_b = this.fixVersion(b)
+          if (ver_a === null) {
+            return 1
+          }
+          if (ver_b === null) {
+            return -1
+          }
+          return gt(new SemVer(ver_a), new SemVer(ver_b)) === true ? -1 : 1
+        },
+      )
+    },
+    runningBeta() {
+      if (this.current_version === null) {
+        return false
+      }
+      return this.current_version.tag.includes('beta')
+    },
+    newBetaAvailable() {
+      if (!this.current_version) {
+        return ''
+      }
+      if (this.latest_beta !== undefined && this.latest_beta !== this.current_version.tag) {
+        return this.latest_beta
+      }
+      return ''
+    },
+    newStableAvailable() {
+      if (!this.current_version) {
+        return ''
+      }
+      if (this.latest_stable !== undefined && this.latest_stable !== this.current_version.tag) {
+        return this.latest_stable
+      }
+      return ''
+    },
+    getLatestBeta() {
+      const ordered_list = this.sortVersions(
+        this.available_versions.remote.map((image) => image.tag)
+          .filter((tag) => this.isSemVer(tag) && tag.includes('beta')),
+      )
+      return ordered_list ? ordered_list[0] : undefined
+    },
+    getLatestStable() {
+      const ordered_list = this.sortVersions(
+        this.available_versions.remote.map((image) => image.tag)
+          .filter((tag) => this.isSemVer(tag) && !tag.includes('beta')),
```

Comment:
It says 'beta' here, we should make a function to encapsulate the map logic and receive an string as parameters, to use for beta and stable